pkg/controller/deployment/deployment_controller.go

```diff
@@ -174,28 +170,6 @@ func (dc *DeploymentController) updateDeployment(old, cur interface{}) {
 	curD := cur.(*extensions.Deployment)
 	glog.V(4).Infof("Updating deployment %s", oldD.Name)
 	dc.enqueueDeployment(curD)
-	// If the selector of the current deployment just changed, we need to requeue any old
```

Comment:
If we're removing `handleOverlap` we can as well remove the selector update annotation we add when `PrepareForUpdate` in deployment/strategy.go